instrumentation/rocketmq-client-4.8/library/src/main/java/io/opentelemetry/instrumentation/rocketmq/TracingConsumeMessageHookImpl.java

```diff
@@ -27,20 +27,18 @@ public void consumeMessageBefore(ConsumeMessageContext context) {
     if (context == null || context.getMsgList() == null || context.getMsgList().isEmpty()) {
       return;
     }
-    Context traceContext = tracer.startSpan(Context.current(), context.getMsgList());
-    ContextAndScope contextAndScope = new ContextAndScope(traceContext, traceContext.makeCurrent());
-    context.setMqTraceContext(contextAndScope);
+    Context otelContext = tracer.startSpan(Context.current(), context.getMsgList());
```

Comment:
Like similar netty libraries, there's no thread guarantee for before and end. Since there's no downstream instrumentation it's not a big deal anyways

Reply:
What about e.g. user-provided message hooks that want to augment existing span?

Reply:
We will need to provide something like `RocketMqTracing.getOpenTelemetryContext(ConsumeMessageContext)` to let them extract it I guess. In the meantime, if someone had such code using `Context.current()` in `consumeMessageAfter` it would generally be buggy with the chance of `Context.current()` not actually corresponding to the hook's message due to context leakage so this fix is important first.

I hope such hooks will be able to use our instrumentation-api instead of that though.